pkg/registry/batch/job/strategy_test.go

```diff
@@ -709,44 +815,205 @@ func TestJobStrategyWithGeneration(t *testing.T) {
 		},
 		Spec: batch.JobSpec{
 			Selector: nil,
-			Template: validPodTemplateSpec,
-		},
+			Template: api.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: validSelector.MatchLabels,
+				},
+				Spec: api.PodSpec{
+					RestartPolicy: api.RestartPolicyOnFailure,
+					DNSPolicy:     api.DNSClusterFirst,
+					Containers:    []api.Container{{Name: "abc", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: api.TerminationMessageReadFile}},
+				},
+			}},
 	}
 
-	Strategy.PrepareForCreate(ctx, job)
-	errs := Strategy.Validate(ctx, job)
-	if len(errs) != 0 {
-		t.Errorf("Unexpected error validating %v", errs)
+	testcases := map[string]struct {
+		jobName  string
+		warnings string
+	}{
+		"happy path job": {
+			jobName: "myjob2",
+		},
+		"dns invalid name": {
+			warnings: "a DNS label is recommended",
+			jobName:  "my job2",
+		},
+	}
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			job.Name = tc.jobName
+			actualWarnings := Strategy.WarningsOnCreate(ctx, job)
+			if tc.warnings != "" {
```

Comment:
I suggest to using `cmp.Diff(wantWarnings, gotWarnings)` to simplify the asserts.

Reply:
The warning is quite long so I decided to search for a token in the string. It’s probably 4 to 5 sentences long so that seemed excessive to test against

Reply:
I see, I'm ok with keeping as is then. However, I think it will not hurt to assert the warnings completely (especially if this is just a single test case, or even a couple). Surfacing the warnings might be helpful to identify or fix some potential problems with them, like being too verbose.

Reply:
I'll see if I can add it.

Reply:
Actually, after a second thought I think the messages are already surfaced in a dedicated test file. Thus, not asserting on actual messages seems enough. Maybe just assert warnings count, or their presence? Anyway, I think just keep the approach consistent for testing WarningsOnUpdate and WarningsOnCreate.